src/quo/components/overlay/view.cljs

```diff
@@ -2,19 +2,23 @@
   (:require
     [quo.components.overlay.style :as style]
     [react-native.blur :as blur]
-    [react-native.core :as rn]))
+    [react-native.core :as rn]
+    [react-native.safe-area :as safe-area]))
 
 (defn view
-  [{:keys [type container-style]} & children]
-  [rn/view {:style (style/overlay-background type)}
-   (if (= type :shell)
-     [blur/view
-      {:blur-amount   20
-       :blur-radius   25
-       :blur-type     :transparent
-       :overlay-color :transparent
-       :style         style/container}
-      [rn/view {:style (merge style/blur-container container-style)}
-       children]]
-     [rn/view {:style (merge style/container container-style)}
-      children])])
+  [{:keys [type container-style top-inset?]} & children]
+  (let [top-style (when top-inset? {:padding-top (safe-area/get-top)})]
+    [rn/view {:style (style/overlay-background type)}
+     (if (= type :shell)
+       [blur/view
+        {:blur-amount   20
+         :blur-radius   25
+         :blur-type     :transparent
+         :overlay-color :transparent
+         :style         style/container}
+        (into [rn/view
+               {:style (merge style/blur-container top-style container-style)}]
+              children)]
+       (into [rn/view
+              {:style (merge style/container top-style container-style)}]
+             children))]))
```

Comment:
There's an open issue for fixing the overlay component with `into`: https://github.com/status-im/status-mobile/issues/19331

Maybe we can close that issue with this PR, but it might be worth checking other usages of overlay and see if we can remove any unneeded `key` props.

Reply:
i think that's fine to keep keys for already added, it won't hurt I guess

Reply:
added issues to the description, thank you